hooks/hookexecution/execution_test.go

```diff
@@ -0,0 +1,153 @@
+
+	testCases := []struct {
+		description         string
+		hookCode            string
+		privacyConfig       *config.AccountPrivacy
+		inPayloadData       hookstage.ProcessedAuctionRequestPayload
+		expectedPayloadData hookstage.ProcessedAuctionRequestPayload
+	}{
+		{
+			description: "payload should change when userFPD is blocked by activity",
+			hookCode:    "foo",
+			inPayloadData: hookstage.ProcessedAuctionRequestPayload{
+				BidRequest: &openrtb2.BidRequest{
+					User: &openrtb2.User{ID: "test_user_id"},
+				},
+			},
+			privacyConfig: getTransmitUFPDActivityConfig("foo", false),
+			expectedPayloadData: hookstage.ProcessedAuctionRequestPayload{
+				BidRequest: &openrtb2.BidRequest{
+					User: &openrtb2.User{ID: ""},
+				},
+			},
+		},
+		{
+			description: "payload should not change when userFPD is not blocked by activity",
+			hookCode:    "foo",
+			inPayloadData: hookstage.ProcessedAuctionRequestPayload{
+				BidRequest: &openrtb2.BidRequest{
+					User: &openrtb2.User{ID: "test_user_id"},
+				},
+			},
+			privacyConfig: getTransmitUFPDActivityConfig("foo", true),
+			expectedPayloadData: hookstage.ProcessedAuctionRequestPayload{
+				BidRequest: &openrtb2.BidRequest{
+					User: &openrtb2.User{ID: "test_user_id"},
+				},
+			},
+		},
+	}
+	for _, test := range testCases {
+		activityControl, err := privacy.NewActivityControl(test.privacyConfig)
+		assert.NoError(t, err)
+		t.Run(test.description, func(t *testing.T) {
+			newPayload := runHandleModuleActivities(test.hookCode, activityControl, test.inPayloadData)
+			assert.Equal(t, test.expectedPayloadData, newPayload)
+		})
+	}
+}
+
+func TestHandleModuleActivitiesNoBidderRequestPayload(t *testing.T) {
+
+	testCases := []struct {
+		description         string
+		hookCode            string
+		privacyConfig       *config.AccountPrivacy
+		inPayloadData       hookstage.RawAuctionRequestPayload
+		expectedPayloadData hookstage.RawAuctionRequestPayload
+	}{
+		{
+			description:         "payload should change when userFPD is blocked by activity",
+			hookCode:            "foo",
+			inPayloadData:       hookstage.RawAuctionRequestPayload{},
+			privacyConfig:       getTransmitUFPDActivityConfig("foo", false),
+			expectedPayloadData: hookstage.RawAuctionRequestPayload{},
+		},
+		{
+			description:         "payload should not change when userFPD is not blocked by activity",
+			hookCode:            "foo",
+			inPayloadData:       hookstage.RawAuctionRequestPayload{},
+			privacyConfig:       getTransmitUFPDActivityConfig("foo", true),
+			expectedPayloadData: hookstage.RawAuctionRequestPayload{},
+		},
+	}
+	for _, test := range testCases {
+		activityControl, err := privacy.NewActivityControl(test.privacyConfig)
+		assert.NoError(t, err)
+		t.Run(test.description, func(t *testing.T) {
+			newPayload := runHandleModuleActivities(test.hookCode, activityControl, test.inPayloadData)
+			assert.Equal(t, test.expectedPayloadData, newPayload)
+		})
+	}
+}
+
```

Comment:
Perhaps a test to insure the original request remains unmolested after the payload has been modified for the module?

Reply:
Yes, this is what is checked in hooks/hookexecution/executor_test.go where tests are added for the functions `ExecuteRawAuctionStage` and `ExecuteBidderRequestStage`. The only difference from other tests there is that `Activities` with the restricted userFPD option is passed.